README.md

```diff
@@ -279,7 +279,19 @@ with pub as (
                     case when bool_or(pubtruncate) then 'truncate' else null end
                 ]) act(name_)
         ) w2j_actions,
-        string_agg(cdc.quote_wal2json(prrelid::regclass), ',') w2j_add_tables
+        case
+            -- collect all tables
+            when bool_and(puballtables) then (
+                select
+                    string_agg(cdc.quote_wal2json((schemaname || '.' || tablename)::regclass), ',')
+                from
+                    pg_tables
```

Comment:
@olirice i think this grabs all the tables, minus the ones belonging to the excluded schemas, and not all the tables belonging to a publication

Reply:
@olirice i've been playing around with the following query in realtime and would love your thoughts:

```sql
with pub as (
        select
          (
            select concat_ws(
                ',',
                case when bool_or(pubinsert) then 'insert' else null end,
                case when bool_or(pubupdate) then 'update' else null end,
                case when bool_or(pubdelete) then 'delete' else null end
              )
            from pg_publication
            where pubname = 'supabase_realtime'
          ) w2j_actions,
          string_agg(realtime.quote_wal2json(format('%I.%I', schemaname, tablename)::regclass), ',') w2j_add_tables
        from
          pg_publication_tables
        where
          pubname = 'supabase_realtime'
        group by
          pubname
        limit 1
      )
      select
        xyz.wal,
        xyz.is_rls_enabled,
        xyz.users,
        xyz.errors
      from
        pub,
        lateral (
          select
            *
          from
            pg_logical_slot_get_changes(
              'realtime', null, null,
              'include-pk', '1',
              'include-transaction', 'false',
              'include-timestamp', 'true',
              'write-in-chunks', 'true',
              'format-version', '2',
              'actions', coalesce(pub.w2j_actions, ''),
              'add-tables', pub.w2j_add_tables
            )
          ) w2j,
          lateral (
            select
              x.wal,
              x.is_rls_enabled,
              x.users,
              x.errors
            from
              realtime.apply_rls(
                wal := w2j.data::jsonb,
                max_record_bytes := 1048576
              ) x(
                wal,
                is_rls_enabled,
                users,
                errors
              )
          ) xyz
      where coalesce(pub.w2j_add_tables, '') <> ''
```

Reply:
I believe this would fail if the publication is enabled for all tables because `pg_publication_tables` would be empty (there's a flag for all tables on `pg_publication` instead)

Reply:
> format('%I.%I', schemaname, tablename)::regclass

yes, this is better.  good call.

Here's a test case where concatenation fails but format succeeds

```sql
create table public."ABC def" (id serial primary key);

set search_path = '';

select 
	(schemaname || '.' || tablename)::regclass, -- will fail
	format('%I.%I', schemaname, tablename)::regclass -- will pass
from pg_tables
where schemaname = 'public'
```

Reply:
I checked the remaining src and all casts to regclass are appropriately wrapped with `quote_ident` which is equivalent to `format('%I', xyz)`.

Pre-coffee bugs :(
Glad you caught it!

Reply:
@olirice 

> I believe this would fail if the publication is enabled for all tables because pg_publication_tables would be empty (there's a flag for all tables on pg_publication instead)

I think that's `pg_publication_rel`. `pg_publicatino_tables` will even give you all the tables in publication defined as `FOR ALL TABLES`.

Reply:
nice one! didn't notice the table name change.

pg_publication_tables looks perfect & concat_ws is a great cleanup too

update to follow